platform/android/MapboxGLAndroidSDK/src/test/java/org/maplibre/android/location/engine/MapLibreFusedLocationEngineImplTest.kt

```diff
@@ -0,0 +1,164 @@
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.junit.MockitoJUnitRunner
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+
+@RunWith(MockitoJUnitRunner::class)
+class MapLibreFusedLocationEngineImplTest {
+    @Mock
+    private val locationManagerMock: LocationManager? = null
+    private var engine: LocationEngine? = null
+    private var maplibreFusedLocationEngineImpl: MapLibreFusedLocationEngineImpl? = null
+    @Before
+    fun setUp() {
+        val context = Mockito.mock(Context::class.java)
+        Mockito.`when`(context.getSystemService(Context.LOCATION_SERVICE)).thenReturn(locationManagerMock)
+        maplibreFusedLocationEngineImpl = MapLibreFusedLocationEngineImpl(context)
+        engine = LocationEngineProxy(maplibreFusedLocationEngineImpl)
+    }
+
+    // J2K: rewrite test to not use "@get"
+    @Test
+    fun getLastLocation() {
+            val latch = CountDownLatch(1)
+            val resultRef = AtomicReference<LocationEngineResult>()
+            val callback = getCallback(resultRef, latch)
+            val location = getMockLocation(LATITUDE, LONGITUDE)
+            val expectedResult = getMockEngineResult(location)
+            Mockito.`when`(locationManagerMock!!.allProviders).thenReturn(mutableListOf("gps", "network"))
+            Mockito.`when`(locationManagerMock.getLastKnownLocation(ArgumentMatchers.anyString())).thenReturn(location)
+            engine!!.getLastLocation(callback)
+            TestCase.assertTrue(latch.await(5, TimeUnit.SECONDS))
+            val result = resultRef.get()
+            Assertions.assertThat(result.lastLocation).isEqualTo(expectedResult.lastLocation)
+    }
+
+    @Test
+    fun createListener() {
+        // J2K: IDE suggestion "as LocationEngineCallback<LocationEngineResult>"
+        val callback: LocationEngineCallback<LocationEngineResult> = Mockito.mock(LocationEngineCallback::class.java) as LocationEngineCallback<LocationEngineResult>
+        val locationListener = maplibreFusedLocationEngineImpl!!.createListener(callback)
+        val mockLocation = getMockLocation(LATITUDE, LONGITUDE)
+        locationListener.onLocationChanged(mockLocation)
+        val argument = ArgumentCaptor.forClass(LocationEngineResult::class.java)
+        Mockito.verify(callback).onSuccess(argument.capture())
+        val result = argument.value
+        Assertions.assertThat(result.lastLocation).isSameAs(mockLocation)
+    }
+
+    @Test
+    fun requestLocationUpdatesOutdoors() {
+        val request = LocationEngineRequest.Builder(10)
+                .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY).build()
+        // J2K: IDE suggestion "as LocationEngineCallback<LocationEngineResult>"
+        val callback: LocationEngineCallback<LocationEngineResult> = Mockito.mock(LocationEngineCallback::class.java) as LocationEngineCallback<LocationEngineResult>
```

Comment:
Here and in other places: is the cast needed?

If not, can you remove the explicit type annotation?

Reply:
a code-err was flagged here by the IDE, followed the suggestions, the cast finally worked.